literals/literals.go

```diff
@@ -0,0 +1,240 @@
+}
+
+func returnStmt(result ast.Expr) *ast.ReturnStmt {
+	return &ast.ReturnStmt{
+		Results: []ast.Expr{result},
+	}
+}
+
+func isTypeDefStr(typ types.Type) bool {
+	strType := types.Typ[types.String]
+
+	if named, ok := typ.(*types.Named); ok {
+		return types.Identical(named.Underlying(), strType)
+	}
+
+	return false
+}
+
+func containsTypeDefStr(expr ast.Expr, info *types.Info) bool {
+	typ := info.TypeOf(expr)
+	if sig, ok := typ.(*types.Signature); ok {
+		for i := 0; i < sig.Params().Len(); i++ {
+			if isTypeDefStr(sig.Params().At(i).Type()) {
+				return true
+			}
+		}
+	}
+
+	if mapT, ok := typ.(*types.Map); ok {
+		return isTypeDefStr(mapT.Elem()) || isTypeDefStr(mapT.Key())
+	}
+
+	if named, ok := typ.(*types.Named); ok {
+		return isTypeDefStr(named)
+	}
+
+	return false
+}
+
+// Obfuscate replace literals with obfuscated lambda functions
+func Obfuscate(files []*ast.File, info *types.Info, fset *token.FileSet, blacklist map[types.Object]struct{}) []*ast.File {
+	pre := func(cursor *astutil.Cursor) bool {
+		switch x := cursor.Node().(type) {
+		case *ast.ValueSpec:
+			return !containsTypeDefStr(x.Type, info)
+
+		case *ast.AssignStmt:
+			for _, expr := range x.Lhs {
+				if index, ok := expr.(*ast.IndexExpr); ok {
+					return !containsTypeDefStr(index.X, info)
+				}
+
+				if ident, ok := expr.(*ast.Ident); ok {
+					return !containsTypeDefStr(ident, info)
+				}
+			}
+		case *ast.CallExpr:
+			return !containsTypeDefStr(x.Fun, info)
+
+		case *ast.CompositeLit:
+			if t, ok := x.Type.(*ast.MapType); ok {
+				return !(containsTypeDefStr(t.Key, info) || containsTypeDefStr(t.Value, info))
+			}
+
+		case *ast.FuncDecl:
+			if x.Type.Results == nil {
+				return true
+			}
+			for _, result := range x.Type.Results.List {
+				for _, name := range result.Names {
+					return !containsTypeDefStr(name, info)
+				}
+			}
+
+		case *ast.KeyValueExpr:
+			if ident, ok := x.Key.(*ast.Ident); ok {
+				return !containsTypeDefStr(ident, info)
+			}
+		case *ast.GenDecl:
+			if x.Tok != token.CONST {
+				return true
+			}
+			for _, spec := range x.Specs {
+				spec, ok := spec.(*ast.ValueSpec)
+				if !ok {
+					return false
+				}
+
+				for _, name := range spec.Names {
+					obj := info.ObjectOf(name)
+
+					// The object itself is blacklisted, e.g. a value that needs to be constant
+					if _, ok := blacklist[obj]; ok {
+						return false
+					}
+				}
+
+				for _, val := range spec.Values {
+					if _, ok := val.(*ast.BasicLit); !ok {
+						return false // skip the block if it contains non basic literals
+					}
+				}
+
+			}
+
+			x.Tok = token.VAR
+			// constants are not possible if we want to obfuscate literals, therefore
+			// move all constant blocks which only contain strings to variables
+
+		}
+		return true
+	}
+
+	post := func(cursor *astutil.Cursor) bool {
+		switch x := cursor.Node().(type) {
+		case *ast.CompositeLit:
+			if info.TypeOf(x.Type).String() == "[]byte" {
```

Comment:
invert and break to unindent the rest

Reply:
also, don't do string comparisons for types; here, you could easily check that the type is Slice, and the Underlying is Byte, etc.